### PR TITLE
BaSP-TG-56: Super admins redux create, update and delete

### PR DESCRIPTION
**Pull request by mara666**
### What has developed?

**- Create _constants_ to:**
  - create an admin
  - update an admin
  - delete an admin
  - get by id an admin

**- Create _actions_ to:**
  - create an admin
  - update an admin
  - delete an admin
  - get by id an admin

**- Create _thunks_ to:**
  - create an admin
  - update an admin
  - delete an admin
  - get by id an admin

**- Complete superAdmins _reducer_ to:**
  - create an admin
  - update an admin
  - delete an admin
  - get by id an admin

**- Connect _superAdmins component_ with store to:**
  - create an admin
  - update an admin
  - delete an admin
  - get by id an admin

### How to test?:
To test this ticket we should open the link of the app deployed in Vercel. Then, we should verify the correct functionality of:

- In the super admins table, the data has to be retrieved correctly and fill the table with all data Admins.
- In the super admins table, when clicking the delete button of a row, there should be a confirmation modal to delete a super admin. In case of 'No', modal should be closed and in case of 'Yes', it should delete the selected admin.
- When clicking on the edit button inside super admins table row, it should navigate to the super admins form page. We should check that the form gets all the admin info and we should update all the changes we made. Once confirmed all these changes, it should navigate back to the super admins table and it should reflect all changes made in the admin who was selected.
- When clicking on the add button in the super admins table page, it should navigate to the super admins form page. We should complete the form with data and press the confirm button. It should navigate to the super admins table and a new row of admin should appear, with all new data.

The main objective of this test is to verify the correct functionality of the app **_after redux implementation_**.

### CODE REVIEW

File: src/Components/SuperAdmins/Form/Form.jsx [hunk 1]
```diff
@@ -15,28 +17,35 @@ const Form = () => {
...
+        email: admins.email,
+        password: admins.password,
+        status: admins.status
+      });
+    }
+  }, [admins]);
+
+  useEffect(() => {
+    if (request === 'POST' || request === 'PUT') {
+      toggleModal(status);
```
Comment (Juanoauil) [inline src/Components/SuperAdmins/Form/Form.jsx:45]: you shouldn't use toggleModal before initialization

File: src/redux/super-admins/reducer.js [hunk 2]
```diff
@@ -1,29 +1,133 @@
...
 
 const superAdminReducer = (state = INITIAL_STATE, action) => {
   switch (action.type) {
     case GET_ADMINS_PENDING:
       return {
         ...state,
-        isPending: true
+        isPending: true,
+        status: 'pending',
+        request: 'GET'
```
Comment (Juanoauil) [inline src/redux/super-admins/reducer.js:34]: you shouldn't set request in redux

File: src/Components/SuperAdmins/Form/Form.jsx [hunk 3]
```diff
@@ -15,28 +17,35 @@ const Form = () => {
...
-        });
-      } catch (error) {
-        alert('Admin does not exist');
-      }
+      dispatch(getAdminsById(idAdmin));
     }
   }, []);
 
+  useEffect(() => {
+    if (request === 'GETBYID') {
```
Comment (Juanoauil) [inline src/Components/SuperAdmins/Form/Form.jsx:32]: remove unnecessary condition

**Review by Juanoauil — CHANGES_REQUESTED** [review]

File: src/Components/SuperAdmins/Form/Form.jsx (hunk 1 above)
Comment (Juanoauil) [reply src/Components/SuperAdmins/Form/Form.jsx:45]: please move toggleModal into onSubmit

Comment (mara666) [reply src/Components/SuperAdmins/Form/Form.jsx:45]: Fixed!

File: src/redux/super-admins/reducer.js (hunk 2 above)
Comment (mara666) [reply src/redux/super-admins/reducer.js:34]: sorry, I removed status and request

File: src/Components/SuperAdmins/Form/Form.jsx (hunk 3 above)
Comment (mara666) [reply src/Components/SuperAdmins/Form/Form.jsx:32]: removed!

**Comment by cami-diaz** [comment]
Hello Dami! When editing a superAdmin and confirming the new data entered, I get this attached modal. I suggest that the message is "SuperAdmin was edited successfully "

image

**Review by Juanoauil — APPROVED** [review]

**Review by DamianPalavecino — COMMENTED** [review]
Replace `status: ''` with `status: false` in line 13 of Form.jsx to fix the issue "Error: status must be a boolean"

**Review by PilarFernandezC — APPROVED** [review]

**Comment by mara666** [comment]
Fixed all issues!

**Pull request merged**